manticore/core/cpu/x86.py

```diff
@@ -4594,6 +4594,17 @@ def PMINUB(cpu, dest, src):
             result |= Operators.ITEBV(dest.size, itema < itemb, itema, itemb) << pos
         dest.write(result)
 
+    @instruction
+    def PMAXUB(cpu, dest, src):
+        dest_value = dest.read()
```

Comment:
```suggestion
        """
        PMAXUB: returns maximum of packed unsigned byte integers in the dest operand
        """
        dest_value = dest.read()
```

Reply:
This should also contain some more details.

Reply:
Can we add some example to docstring? E.g.:

```
$xmm1.v16_int8 = {0xe9, 0x44, 0x11, 0x0, 0xee, 0xe2, 0xe2, 0xff, 0xd8, 0xd7, 0xd6, 0xd5, 0xd4, 0xd3, 0xd2, 0xd1}
$xmm2.v16_int8 = {0xe8, 0xe7, 0xe6, 0xe5, 0xe4, 0xe3, 0xe2, 0xe1, 0xf8, 0xf7, 0xf6, 0xf5, 0xf4, 0xf3, 0xf2, 0xf1}

# executes pmauxb xmm1, xmm2
$xmm1.v16_int8 = {0xe9, 0xe7, 0xe6, 0xe5, 0xee, 0xe3, 0xe2, 0xff, 0xf8, 0xf7, 0xf6, 0xf5, 0xf4, 0xf3, 0xf2, 0xf1}
```

JFYI I tested the instruction using gdb+pwndbg and pwntools. I made such Python script:
```python
from pwn import *

r = debug_assembly('''
    mov rax, 0xFFE2E2EE001144E9
    movq [rsp], rax
    mov rax, 0xD1D2D3D4D5D6D7D8
    movq [rsp+8], rax
    mov rax, 0xE1E2E3E4E5E6E7E8
    movq [rsp+16], rax
    mov rax, 0xF1F2F3F4F5F6F7F8
    movq [rsp+24], rax
    movdqu xmm1, [rsp]
    movdqu xmm2, [rsp+16]
    nop
    pmaxub xmm1, xmm2
''', arch='amd64')
r.interactive()
```

launched it (under tmux so it opened another console with gdb that also loaded pwndbg) and then went through insrtructions by `si`, made `set print elements 0` and printed the values by e.g. `p/x $xmm1.v16_int8`.

Reply:
We can also change the description to e.g. include link/some info from [Intel's architectures software developer instruction set reference manual](https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-instruction-set-reference-manual-325383.pdf) in the `PMAXUB/PMAXUW—Maximum of Packed Unsigned Integers`'s description, or to be more specific - page 964 (emphasis mine):


> Performs a SIMD compare of the packed unsigned byte, word integers in the second source operand and the first source operand and returns the maximum value for each pair of integers to the destination operand.
> Legacy SSE version PMAXUB: The source operand can be an MMX technology register or a 64-bit memory location. The destination operand can be an MMX technology register.
>
> 128-bit Legacy SSE version: The first source and destination operands are XMM registers. The second source
operand is an XMM register or a 128-bit memory location. Bits (MAX_VL-1:128) of the corresponding destination register remain unchanged.
> 
> VEX.128 encoded version: The first source and destination operands are XMM registers. The second source
operand is an XMM register or a 128-bit memory location. Bits (MAX_VL-1:128) of the corresponding destination register are zeroed.
> 
> VEX.256 encoded version: The second source operand can be an YMM register or a 256-bit memory location. The first source and destination operands are YMM registers.
>
> EVEX encoded versions: The first source operand is a ZMM/YMM/XMM register; The second source operand is a ZMM/YMM/XMM register or a 512/256/128-bit memory location. The destination operand is conditionally updated based on writemask k1.